packages/flutter_tools/lib/src/test/coverage_collector.dart

```diff
@@ -204,20 +204,17 @@ Future<Map<String, dynamic>> collect(Uri serviceUri, bool Function(String) libra
   int i = 0;
   while (i < kPollAttempts) {
```

Comment:
Style guide says we should just spin forever.

Reply:
That would put us in a state where a single misbehaving test would remove our ability to collect coverage. If we were starting from scratch, I agree we should spin forever here. But the tools tests aren't currently stable or hermetic enough where we can afford to do so. I'm concerned that would require us to more or less give up on collecting coverage in the short-medium term.

I'd like to start applying some of the techniques the WidgetTester uses to prevent leaking Timers/Futures gradually via https://github.com/flutter/flutter/pull/35392 though

Reply:
Would `debugName` provide any insight into which test is misbehaving?